Add debiased distribution compression source code

lingxiaoli94 opened the pull request.

lingxiaoli94 committed:
add source code for debiased distribution compression

lingxiaoli94 committed:
add covtype gold hmc samples for the approximate mcmc experiment

lingxiaoli94 commented:
@microsoft-github-policy-service agree [company="MIT"]

lingxiaoli94 commented:
@microsoft-github-policy-service agree

lmackey reviewed (changes requested):
I'd recommend using a persistent data repository like https://dataverse.harvard.edu/ for hosting this data.

lmackey commented on goodpoints/jax/README.md:
```diff
@@ -0,0 +1,31 @@
+This package contains algorithms from Debiased Distribution Compression by Li et al. 2024 implemented in JAX.
+
+## Debiased distribution compression algorithms
+The `dtc.py` module contains the meta debiased-then-compress algorithm for debiased compression along with 6 instantiations corresponding to the 6 algorithms in the paper.
```
debias-then-compress

lmackey commented on goodpoints/jax/README.md:
```diff
@@ -0,0 +1,31 @@
+The `serial/` package provides utilities for serialization and caching of intermediate outputs.
+
+## Dependencies
+The main dependency is JAX whose install instructions can be found at [here](https://jax.readthedocs.io/en/latest/quickstart.html).
```
The main dependency is JAX; the install instructions can be found

lmackey commented on README.md:
````diff
@@ -7,12 +7,15 @@ GoodPoints is a collection of tools for compressing a distribution more effectiv
 ```
 pip install goodpoints
 ```
+For **Debiased Distribution Compression**, JAX and additional independencies are required. See [goodpoints/jax](goodpoints/jax) for more details.
````
dependencies

lmackey commented on README.md:
```diff
@@ -7,12 +7,15 @@ GoodPoints is a collection of tools for compressing a distribution more effectiv
 - Given an initial summary of n input points, **kernel thinning** returns s << n output points with comparable integration error across a reproducing kernel Hilbert space
 - **Compress++** reduces the runtime of generic thinning algorithms with minimal loss in accuracy
 - **Compress Then Test** accelerates kernel two-sample testing using high-fidelity compression
+- **Debiased Distribution Compression** provides a suite of JAX-accelerated methods for debiasing and compressing input points, supporting equal-weighted, simplex-weighted, and constant-preserving coresets
```
I would remove the Debiased Distribution Compression bullet and instead add the following bullet:
- **Stein Kernel Thinning**, **Stein Recombination**, and **Stein Cholesky** deliver unweighted,  simplex-weighted, and constant-preserving coresets that reduce biases due to off-target sampling, tempering, or burn-in

lmackey commented on README.md:
````diff
@@ -164,6 +175,17 @@ by executing `examples/kt/run_kt_experiment.ipynb` with appropriate parameters.
 
+#### [Debiased Distribution Compression](https://arxiv.org/pdf/2404.12290)
+```
+@article{li2024debiased,
````
InProceedings{li2024debiased,
  title = 	 {Debiased Distribution Compression},
  author =       {Li, Lingxiao and Dwivedi, Raaz and Mackey, Lester},,
  booktitle = 	 {Proceedings of the 41st International Conference on Machine Learning},
  year = 	 {2024},
  volume = 	 {203},
  series = 	 {Proceedings of Machine Learning Research},
  month = 	 {21--27 Jul},
  publisher =    {PMLR},
}

lingxiaoli94 committed:
update READMEs

lingxiaoli94 committed:
Add Harvard dataverse link

lingxiaoli94 commented:
I have addressed the comments and added the link to Harvard dataverse at [here](https://dataverse.harvard.edu/dataset.xhtml?persistentId=doi:10.7910/DVN/NGEM9H).

lmackey commented:
Adding debiased distribution compression code
